src/citrine/resources/data_concepts.py

```diff
@@ -544,73 +542,14 @@ def register_all(self,
-        scope = temp_scope if dry_run else CITRINE_SCOPE
-        set_uuids(models, scope=scope)
-
-        resources = list()
-        batch_size = 50
-        result_index = dict()
-        if dry_run:
-            batcher = Batcher.by_dependency()
-        else:
-            batcher = Batcher.by_type()
-
-        if status_bar:
-            desc = "Verifying GEMDs" if dry_run else "Registering GEMDs"
-            iterator = tqdm(batcher.batch(models, batch_size), leave=False, desc=desc)
-        else:
-            iterator = batcher.batch(models, batch_size)
-
-        for batch in iterator:
-            objects = [replace_objects_with_links(scrub_none(model.dump())) for model in batch]
-            response_data = self.session.put_resource(
-                path + '/batch',
-                json={'objects': objects},
-                params=params
-            )
-            registered = [self.build(obj) for obj in response_data['objects']]
-            result_index.update(make_index(registered))
-            substitute_objects(registered, result_index, inplace=True)
-
-            if not dry_run:
-                # Platform may add a CITRINE_SCOPE uid and citr_auto tags; update locals
-                for obj in batch:
-                    result = result_index[obj.to_link()]
-                    obj.uids.update({k: v for k, v in result.uids.items()})
-                    if result.tags is not None:
-                        obj.tags = list(result.tags)
-            else:
-                # Remove of the tags/uids the platform spuriously added
-                # this might leave objects with just the temp ids, which we want to strip later
-                for obj in batch:
-                    result = result_index[obj.to_link()]
-                    if CITRINE_SCOPE not in obj.uids:
-                        citr_id = result.uids.pop(CITRINE_SCOPE, None)
-                        result_index.pop(LinkByUID(scope=CITRINE_SCOPE, id=citr_id), None)
-                    if result.tags is not None:
-                        todo = [tag for tag in result.tags
-                                if re.match(f"^{CITRINE_TAG_PREFIX}::", tag)]
-                        for tag in todo:  # Covering this block would require dark art
-                            if tag not in obj.tags:
-                                result.tags.remove(tag)
-
-            resources.extend(registered)
-
-        if dry_run:  # No-op if not dry-run
-            recursive_foreach(list(models) + list(resources),
-                              lambda x: x.uids.pop(temp_scope, None))  # Strip temp uids
-        return resources
+        from citrine.resources.gemd_resource import GEMDResourceCollection
+        gemd_collection = GEMDResourceCollection(self.project_id, self.dataset_id, self.session)
+        return gemd_collection.register_all(
+            models,
+            dry_run=dry_run,
+            status_bar=status_bar,
+            include_nested=include_nested
+        )
```

Comment:
All DataConcepts register_all calls should route through GEMDResourceCollection / `storables`